### PR TITLE
ID-55 select 2

### PR DESCRIPTION
**Pull request by johnhuangclever**
**Overview:**
Single-item select which is a text-searchable drop down. Built using downshift's useComboBox hook

Unfinished Select 2.0 Dewey spec here:
https://www.figma.com/file/IBVNvo2dI221k4Dye4Hg0S/Dewey-2.0-WIP?node-id=21490%3A11194 

Interaction modelled after material ui drop down https://material-ui.com/components/autocomplete

**Screenshots/GIFs:**
ID-55-select-2-input

**Testing:**

- [x] Unit tests
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [ ] IE11

**Roll Out:**

- Before merging:
  - [x] Updated docs
  - [ ] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [x] Add an entry in `ComponentsView.componentsToDisplay` using this template:
      ```
      {
        componentLink: "<COMPONENT LINK>",
        componentImg: "<COMPONENT LINK>.png",
        componentName: "<COMPONENT NAME>",
        componentImgAlt: "A <COMPONENT NAME> component",
      },
      ```
    - [x] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/Select2/Select2.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                className={classNames(
+                  cssClass.MENU_OPTION,
+                  i === highlightedIndex && cssClass.MENU_OPTION_HIGHLIGHTED,
+                )}
+                key={`${o.value}${i}`}
+                {...getItemProps({
+                  item: o,
+                  index: i,
+                  onClick: () => {
+                    console.log("clicking!");
```
Comment (prime-time) [inline src/Select2/Select2.tsx:194]: ✂️

File: src/Select2/Select2.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+          {...getInputProps({
+            ref: inputRef,
+            onClick: (e) => {
+              if (!isOpen) {
+                openMenu();
+                if (inputRef.current) {
+                  inputRef.current.select();
+                }
+              }
+            },
```
Comment (prime-time) [inline src/Select2/Select2.tsx:153]: It looks like this callback function is the same as the onClick handler a few lines above, so you could reuse the same function if you wanted to save a couple of lines of code. It's not quite at the "rule of three" threshold, so feel free to ignore this.

**Review by prime-time — APPROVED** [review]
Nice, this is looking great so far 🏎️ 💨

File: src/Select2/Select2.tsx (hunk 2 above)
Comment (johnhuangclever) [reply src/Select2/Select2.tsx:153]: Turns out I don't think I need the one on the `<input>`. The parent click handler is enough 👍

**Pull request merged**